consensus/parlia/snapshot.go

```diff
@@ -410,6 +461,24 @@ func parseValidators(header *types.Header, chainConfig *params.ChainConfig, parl
 	return cnsAddrs, voteAddrs, nil
 }
 
+func parseTurnTerm(header *types.Header, chainConfig *params.ChainConfig, parliaConfig *params.ParliaConfig) (*uint8, error) {
+	if header.Number.Uint64()%parliaConfig.Epoch != 0 ||
+		!chainConfig.IsBohr(header.Number, header.Time) {
+		return nil, nil
+	}
+
+	if len(header.Extra) <= extraVanity+extraSeal {
+		return nil, errors.New("invalid turnTerm")
+	}
+	num := int(header.Extra[extraVanity])
+	pos := extraVanity + validatorNumberSize + num*validatorBytesLength
+	if len(header.Extra) <= pos {
+		return nil, errors.New("invalid turnTerm")
+	}
+	turnterm := header.Extra[pos]
```

Comment:
How to distinguish the empty `turnterm`? Here is the extra structure comment.

```golang
// |---Extra Vanity---|---Validators Number and Validators Bytes (or Empty)---|---Turn Term (or Empty)---|---Vote Attestation (or Empty)---|---Extra Seal---|
```

Reply:
> How to distinguish the empty `turnterm`? Here is the extra structure comment.
> 
> ```go
> // |---Extra Vanity---|---Validators Number and Validators Bytes (or Empty)---|---Turn Term (or Empty)---|---Vote Attestation (or Empty)---|---Extra Seal---|
> ```

If IsBohr(header.Number, header.Time), the `turnterm` won't be empty, otherwise will be empty

Reply:
Ok, I get it, maybe the comment below will be more accurate.
```golang
// |---Extra Vanity---|---Validators Number & Validators Bytes & Turn Term (or Empty)---|---Vote Attestation (or Empty)---|---Extra Seal---|
```
It cannot be empty in epoch change header after `Bohr`, like validator set, otherwise it will be empty.